indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord.supervisor;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import io.druid.indexing.overlord.TaskMaster;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Endpoints for submitting and starting a {@link SupervisorSpec}, getting running supervisors, stopping supervisors,
+ * and getting supervisor history.
+ */
+@Path("/druid/indexer/v1/supervisor")
+public class SupervisorResource
+{
+  private final TaskMaster taskMaster;
+
+  @Inject
+  public SupervisorResource(TaskMaster taskMaster)
+  {
+    this.taskMaster = taskMaster;
+  }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response specPost(final SupervisorSpec spec)
+  {
+    return asLeaderWithSupervisorManager(
+        new Function<SupervisorManager, Response>()
+        {
+          @Override
+          public Response apply(SupervisorManager manager)
+          {
+            if (manager.hasSupervisor(spec.getId())) {
+              return Response.status(Response.Status.CONFLICT)
```

Comment:
ok, so if user needs to make some change to spec e.g. changing the schema, then they are supposed to first shutdown the supervisor and then do the POST.
this one should be in the doc i believe.can you please add a section in the doc about how to make changes, also if user is supposed to do anything special as part of roling deployment.